### PR TITLE
doc:adding doc for pocketlab

### PR DESCRIPTION
**Pull request by D-Disha**
Added documentation for pocketlab
Added pocketlab.rst ,pl_gui_views.rst
and pl_firmware_architecture.rst.
Added Pocketlab section to IIO Tools section.

### CODE REVIEW

**Comment by D-Disha** [comment]
Hi all, are there any comments on this PR?

File: doc/sphinx/source/pocketlab/pl_gui_views.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Register View
+-------------
+This view provides low level access to the registers of the device. The user can read and the register values in hex.
+
+.. image:: /source/pocketlab/register_view.jpg
+   :width: 400
+
+DMM View
+--------
+This view continuously displays device specific ADC read back value in volts, once start button is activated.
```
Comment (jansunil) [inline doc/sphinx/source/pocketlab/pl_gui_views.rst:26]: ADC readback values in "SI Units"

File: doc/sphinx/source/pocketlab/pl_gui_views.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+------------
+In this view adc raw data can be captured and visualized as waveforms continuously once the start button is activated.
+
+.. image:: /source/pocketlab/capture_view.jpg
+   :width: 400
+
+Analysis View
+-------------
+In this view FFT analysis can be visualized. Key information such as SNR, THD, Fund Power, freq and RMS noise is displayed on the right pane.
+FFT analysis can be performed only on one channel at a time.
```
Comment (jansunil) [inline doc/sphinx/source/pocketlab/pl_gui_views.rst:42]: Should we put a note for this line?

**Comment by jansunil** [comment]
@D-Disha Can we also add proper commit and PR description ?

**Review by jansunil — DISMISSED** [review]

**Comment by ribdp** [comment]
nit: Typo in commit description (spelling of pl_firmware)

File: doc/sphinx/source/pocketlab/pocketlab.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
+============
+Introduction
+============
+Pocket lab is a portable display-based embedded solution created primarily for demoing the ADCs and DACs functionality.
```
Comment (ribdp) [inline doc/sphinx/source/pocketlab/pocketlab.rst:4]: Do we have any DAC supported in PL currently?

File: doc/sphinx/source/pocketlab/pl_gui_views.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+Configuration View
+------------------
+Provides ability to read and write the global and channel attributes of the device.
+
+.. image:: /source/pocketlab/config_view.jpg
+   :width: 400
+
+Register View
+-------------
+This view provides low level access to the registers of the device. The user can read and the register values in hex.
```
Comment (SaikiranGudla) [inline doc/sphinx/source/pocketlab/pl_gui_views.rst:19]: Remove `and`

**Review by SaikiranGudla — COMMENTED** [review]
Can correct case of `o` in `POcketlab` word in commit message. Also, shall we either go with `Added` or `Adding` at all places in the commit message and not use both?

File: doc/sphinx/source/pocketlab/pocketlab.rst (hunk 3 above)
Comment (D-Disha) [reply doc/sphinx/source/pocketlab/pocketlab.rst:4]: No will remove this

File: doc/sphinx/source/pocketlab/pocketlab_fw_architecture.png
Comment (ribdp) [inline doc/sphinx/source/pocketlab/pocketlab_fw_architecture.png:1]: What does the Eval board picture signify here? If we need to include it, should we add it as a separate block that the Tinyiiod application talks to?

Comment (D-Disha) [reply doc/sphinx/source/pocketlab/pocketlab_fw_architecture.png:1]: I actually took this image from the confluence page that Mahesh created. I think it just means that the IIO Device is the eval board and hence kept right next to it

File: doc/sphinx/source/pocketlab/pocketlab.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+This documentation describes various components of pocket lab application in subsequent sections.
+
+.. Pocketlab Firmware Structure
+.. include:: /source/pocketlab/pocketlab_firmware_structure.rst
+
+=================
+Pocketlab Library
+=================
+
+Source Code: https://github.com/analogdevicesinc/precision-converters-library/tree/main/pocket_lab
```
Comment (SaikiranGudla) [inline doc/sphinx/source/pocketlab/pocketlab.rst:16]: image
There's a slight misalignment in the above links in the html page. 

```suggestion
Source Code: `Precision Converters Library <https://github.com/analogdevicesinc/precision-converters-library/tree/main/pocket_lab>`_
```

Can do same changes for below line as well.

**Review by jansunil — APPROVED** [review]

**Review by ribdp — APPROVED** [review]

**Review by SaikiranGudla — APPROVED** [review]

**Pull request merged**